### PR TITLE
Add filtering to vocabulary

### PR DESCRIPTION
**Pull request by psoleckimoj**

### CODE REVIEW

File: modules/custom/moj_resources/src/Utilities.php [hunk 1]
```diff
@@ -18,25 +18,35 @@ class Utilities {
     *
     * @param int $prisonId
     * @param int[] $prisonCategories
-    * @param QueryInterface
+    * @param QueryInterface $query
+    * @param bool $isTerm
```
Comment (psoleckimoj) [inline modules/custom/moj_resources/src/Utilities.php:22]: If you try and query on a non-existent field the query blows up, added this switch.

File: modules/custom/moj_resources/src/Utilities.php [hunk 2]
```diff
@@ -93,7 +103,7 @@ public static function getPrisonsFor($node) {
     $prisons = array();
 
     foreach ($node->field_moj_prisons as $prison) {
-      array_push($prisons, $prison->target_id);
+      array_push($prisons, intval($prison->target_id));
```
Comment (psoleckimoj) [inline modules/custom/moj_resources/src/Utilities.php:106]: After finding some of these are coming through as strings when doing the Term API I don't trust these.

Comment (LucasCairns) [reply modules/custom/moj_resources/src/Utilities.php:106]: So this does casting similar to `(int) $prisonCategory->target_id`?

**Review by LucasCairns — APPROVED** [review]
👍

File: modules/custom/moj_resources/src/Utilities.php (hunk 2 above)
Comment (psoleckimoj) [reply modules/custom/moj_resources/src/Utilities.php:106]: Yep, it's basically the same as `Number.parseInt()` in JS.

**Pull request merged**